### ISSUE TITLE
Tests need to be fixed: printing to stdout, docstrings, dict

### ISSUE THREAD

**Issue by diego-plan9** [Qiskit/qiskit-terra#840]
### Information

Running the tests currently results in an extra line being printed directly to stdout using `print()`:

```
$ python -m unittest 
QISKIT_TESTS: {'': True, 'rec': False, 'skip_online': False, 'run_slow': False, 'mock_online': False}
...
```

and additionally the first element of the dict looks a bit suspicious (`'': True`).

@1ucian0, can you take a look at it, removing the `print()` statement or changing it to a log entry, in order to preserve the output of the testing tools as verbatim as possible? Additionally, it seems there are a bunch of docstrings in `common.py` where the types are not correctly set (most of them related to `QISKIT_TESTS` - they indicate a `list` but probably meant to be a `dict`).

**Comment by delapuente** [comment]
The `'': True` is a bug I think. The `print()` was useful to remind the user which options are enabled. I'm happy to change it to a log (which is also included per test-suite, I think), but it turns out that controlling the general log-level for printing to the standard output is not as easy as I expected.

**Comment by diego-plan9** [comment]
We really should avoid modifying the default `stdout` output if possible, as by doing so it becomes impossible to separate between the output of the tools vs. what output we chose to generate in most testing flows - if using logging, `stderr` will instead be used along with usual the fine-grain benefits, which is a more flexible approach. That is part of the reason why we did a removal of `print()` statements a while back, actually!
